src/components/articles/single/singleArticleContent.tsx

```diff
@@ -1,21 +1,26 @@
-import React from 'react';
+import React, { useState } from 'react';
 
 import { Article } from '../../../models/article';
 import { markdown } from '../../../remarkable/remarkable';
 import ArticleContainer from '../../styled-components/articles/ArticleContainer';
 import TaskCard from '../../tasks/single/taskCard';
 import LinkedTasksContainer from '../../styled-components/articles/LinkedTasksContainer';
 import { taskComparator, TaskStatus } from '../../../models/task';
+import TaskModal from '../../tasks/single/taskModal';
 
 interface SingleArticleProps {
     article: Article;
 }
 const statusOrder = [TaskStatus.OK, TaskStatus.BLOCKED, TaskStatus.SOLVED];
 
 const SingleArticleContent = ({ article }: SingleArticleProps) => {
+    const [taskId, setTaskId] = useState(0);
```

Comment:
I don't like fact that we have mixed logic for tasks and article here. Maybe it should be split up (separate article and linked tasks)? And also there is duplicated logic between this component and `TaskCategoryComponent`